pkg/health/condition/check_backup_ready.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package condition
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	coordinationv1 "k8s.io/api/coordination/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type backupReadyCheck struct {
+	cl client.Client
+}
+
+const (
+	BackupSucceeded string = "BackupSucceeded"
+	BackupFailed    string = "BackupFailed"
+	Unknown         string = "Unknown"
+)
+
+func (a *backupReadyCheck) Check(ctx context.Context, etcd druidv1alpha1.Etcd) Result {
+	//Default case
+	result := &result{
+		conType: druidv1alpha1.ConditionTypeBackupReady,
+		status:  druidv1alpha1.ConditionUnknown,
+		reason:  Unknown,
+		message: "Cannot determine etcd backup status since no snapshot leases present",
+	}
+
+	//Fetch snapshot leases
+	fullSnapLease := &coordinationv1.Lease{}
+	deltaSnapLease := &coordinationv1.Lease{}
+	var fullSnapErr, incrSnapErr error
+	fullSnapErr = a.cl.Get(ctx, types.NamespacedName{Name: getFullSnapLeaseName(&etcd), Namespace: etcd.ObjectMeta.Namespace}, fullSnapLease)
+	incrSnapErr = a.cl.Get(ctx, types.NamespacedName{Name: getDeltaSnapLeaseName(&etcd), Namespace: etcd.ObjectMeta.Namespace}, deltaSnapLease)
+
+	if fullSnapErr != nil || incrSnapErr != nil || (fullSnapLease.Spec.RenewTime == nil && deltaSnapLease.Spec.RenewTime == nil) {
+		return result
+	}
+
+	deltaLeaseRenewTime := deltaSnapLease.Spec.RenewTime
+	fullLeaseRenewTime := fullSnapLease.Spec.RenewTime
+	fullLeaseCreateTime := &fullSnapLease.ObjectMeta.CreationTimestamp
+
+	if fullLeaseRenewTime == nil && deltaLeaseRenewTime != nil && fullLeaseCreateTime != nil {
+		//Most probable during reconcile of existing clusters if fresh leases are created
+		//Treat backup as succeeded if delta snap lease renewal happens in the required time window and full snap lease not more than 24h old
+		if time.Since(deltaLeaseRenewTime.Time) < 2*etcd.Spec.Backup.DeltaSnapshotPeriod.Duration && time.Since(fullLeaseCreateTime.Time) < 24*time.Hour {
```

Comment:
```suggestion
		if time.Since(deltaLeaseRenewTime.Time) < 1.1*etcd.Spec.Backup.DeltaSnapshotPeriod.Duration && time.Since(fullLeaseCreateTime.Time) < 24*time.Hour {
```
why `2*etcd.Spec.Backup.DeltaSnapshotPeriod.Duration` ? What about `1.1` if we want more to be more precise

Reply:
That's a little bit arbitrary 
2 in the sense that we can tolerate one missed update, but 2 missed updated in a row would be reason to suspect some component is failing

wdyt? would you recommend something else?

Reply:
> 2 in the sense that we can tolerate one missed update, but 2 missed updated in a row would be reason to suspect some component is failing

if that’s the case then it this condition get fails when `time.Since(deltaLeaseRenewTime.Time)=10min` and `2*etcd.Spec.Backup.DeltaSnapshotPeriod.Duration=10min`

Reply:
Right
in this case since since lease had not been updated in 10 mins (2 missed updates) this is expected to fail

Reply:
but what I'm saying is lets suppose `time.Since(deltaLeaseRenewTime.Time)=10min` and deltaSnashot is taken but Snapshotter is about to update the delta lease but before that custodian controller checks lease then isn't this condition can occur.

Reply:
Hm, should we first transition to `Unknown` and then to `False` eventually? This can help to only fire alerts when the condition is `False` and to handle intermittent states as mentioned by @ishan16696 gracefully.

Reply:
Hm, you're right, that will help
I've added the intermediate state in [`f61fb00`](https://github.com/gardener/etcd-druid/pull/271/commits/f61fb002123b5b69fd231ea7057ce2446c709cb2)

Reply:
The changes look good. However, I noticed that we have the same dependency as discussed [here](https://github.com/gardener/etcd-druid/pull/271#discussion_r774607346), right? We actually need [etcd-backup-restore#410](https://github.com/gardener/etcd-backup-restore/pull/410) to be merged and released first. Should we make it a TODO as well or do you have any other suggestion?

Reply:
https://github.com/gardener/etcd-druid/pull/271#issuecomment-1012259584